apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -149,6 +149,18 @@ where
                     continue;
                 }
                 TxType::Protocol(protocol_tx) => match protocol_tx.tx {
+                    #[cfg(not(feature = "abcipp"))]
+                    ProtocolTxType::EthEventsVext(ref ext) => {
+                        for event in ext.data.ethereum_events.iter() {
+                            self.mode.deque_eth_event(event);
```

Comment:
Should we be doing this here? The vext could have come from another validator. We only want to dequeue Ethereum events that we (the validator) have the successfully included in a block, so we should have an additional check here that it is actually our vext.

This issue was already present I think, so should be a separate PR/issue (if my understanding above is correct).

Reply:
https://github.com/anoma/namada/issues/818 opened this issue